jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -16,431 +16,28 @@
-   */
-  @VisibleForTesting
-  static JarType determineJarType(Path jarPath) throws IOException {
-    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
-      if (jarFile.getEntry("BOOT-INF") != null) {
-        return JarType.SPRING_BOOT;
-      }
-      return JarType.STANDARD;
-    }
-  }
-
-  /**
-   * Creates layers for dependencies, snapshot dependencies, resources and classes on container for
-   * a standard jar.
-   *
-   * @param jarPath path to jar file
-   * @param tempDirPath path to temporary jib local directory
-   * @return list of {@link FileEntriesLayer}
-   * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
-   *     provided doesn't exist
-   */
-  static List<FileEntriesLayer> createLayersForExplodedStandard(Path jarPath, Path tempDirPath)
-      throws IOException {
-    // Add dependencies layers.
-    List<FileEntriesLayer> layers = getDependenciesLayers(jarPath, ProcessingMode.exploded);
-
-    // Determine class and resource files in the directory containing jar contents and create
-    // FileEntriesLayer for each type of layer (classes or resources), while maintaining the
-    // file's original project structure.
-    Path localExplodedJarRoot = tempDirPath;
-    ZipUtil.unzip(jarPath, localExplodedJarRoot);
-    Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
-    Predicate<Path> isResourceFile = isClassFile.negate().and(Files::isRegularFile);
-    FileEntriesLayer classesLayer =
-        addDirectoryContentsToLayer(
-            CLASSES, localExplodedJarRoot, isClassFile, APP_ROOT.resolve("explodedJar"));
-    FileEntriesLayer resourcesLayer =
-        addDirectoryContentsToLayer(
-            RESOURCES, localExplodedJarRoot, isResourceFile, APP_ROOT.resolve("explodedJar"));
-
-    if (!resourcesLayer.getEntries().isEmpty()) {
-      layers.add(resourcesLayer);
-    }
-    if (!classesLayer.getEntries().isEmpty()) {
-      layers.add(classesLayer);
-    }
-    return layers;
-  }
-
-  /**
-   * Creates layers for dependencies, snapshot dependencies and the jar itself on container for a
-   * standard jar.
-   *
-   * @param jarPath path to jar file
-   * @return list of {@link FileEntriesLayer}
-   * @throws IOException if I/O error occurs when opening the jar file
-   */
-  static List<FileEntriesLayer> createLayersForPackagedStandard(Path jarPath) throws IOException {
-    // Add dependencies layers.
-    List<FileEntriesLayer> layers = getDependenciesLayers(jarPath, ProcessingMode.packaged);
-
-    // Add layer for jar.
-    FileEntriesLayer jarLayer =
-        FileEntriesLayer.builder()
-            .setName(JAR)
-            .addEntry(jarPath, APP_ROOT.resolve(jarPath.getFileName()))
-            .build();
-    layers.add(jarLayer);
-
-    return layers;
-  }
+/** Interface to create layers and compute entrypoint from jar file contents. */
+public interface JarModeProcessor {
```

Comment:
Unfortunately, I had to modify `createLayers()` and `computeEntrypoint()` to no longer accept parameters for the sake of consistency. In the case of `createLayers()`, while explode mode requires a temporary directory, packaged mode doesn't. On the other hand, `computeEntrypoint()` for spring-boot exploded **doesn't** require the `jarPath` but the same for spring-boot packaged **does**. The `jarPath` and `temporaryDirectory` are now set using setters.